File: 389-find-the-difference/389-find-the-difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        d1,d2={},{}
        for i in t:
            if i in d1:
                d1[i] += 1
            else:
                d1[i] = 1
        for i in s:
            if i in d2:
                d2[i] += 1
            else:
                d2[i] = 1
        for i in d1:
            if i not in d2 or d1[i] != d2[i]:
                return i

File: 389-find-the-difference/389-find-the-difference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        map<char,int> m1,m2;
        char res;
        for(auto i: t){
            if(m1.find(i)!=m1.end()) m1[i]+=1;
            else m1[i] = 1;
        }
        for(auto i: s){
            if(m2.find(i)!=m2.end()) m2[i]+=1;
            else m2[i] = 1;
        }
        for(auto i: m1)
            if(m2.find(i.first)==m2.end() || m1[i.first]!=m2[i.first])
                res = i.first;
        return res;
    }
};